mautic/mautic
# Campaign decision: Focus Item view

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/243
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
New decision make Focus Item more sweat.
Current campaign action https://github.com/mautic/mautic/pull/4666 just insert js code to website. 
Decision fire JS code with every page hit, but is executed as true just if  Focus Item is opened (view pixel event). 

Use case:
Campaign will insert exit popup everytime until  popup  will fired.

![image](https://user-images.githubusercontent.com/462477/33911065-52203436-df91-11e7-84f4-81dc3e4c302f.png)

#### Steps to test this PR:
1.  Create Focus Item - exit popup
2.  Then create campaign with decision - Focus Item show and add action Modify tags after the decision
3: Go to page with tracking code and open developer tools (firebug..)
You should see focus item loaded in DOM
![image](https://user-images.githubusercontent.com/462477/33911282-29529c96-df92-11e7-8658-982c4506f761.png)
4. Go to Campaign and see if Modify tags wasn't executed
5. Then come back to tracking page and go to exit from page. 
6. Popup should be triggered
7. Go to campaign and see if Modify tags action was executed 
8. Repeat tests with Limit to URLS option
